services/auth/source/ldap/source_sync.go

```diff
@@ -178,8 +178,9 @@ func (source *Source) Sync(ctx context.Context, updateExisting bool) error {
 				}
 			}
 
-			if usr.IsUploadAvatarChanged(su.Avatar) {
-				if err == nil && source.AttributeAvatar != "" {
+			if err == nil && source.AttributeAvatar != "" {
```

Comment:
Where is the `err` from? Why checking it?

Reply:
I would suppose all changed code is fully understood by the authors.

Otherwise copy-paste existing code could cause new bugs, even if you didn't change the old code.

----

For this case, maybe it is not that serious, but I still hope we can figure out the correct approach.

Reply:
I did not copy and paste it; it is literally the existing code, which I added 3 other fixes for. I think it takes some understanding of the code to be able to find 3 different issues with a mere handful of lines of code and fix them, no?

I simply didn't expect there to be even more issues than I found, because that's already a number that is too high for production code, and also because I made sure that the actual behavior of the function is fixed. I am not an experienced Go programmer, and there are err checks like this for almost every function, so it looked perfectly reasonable to me.

If you think it doesn't belong there, then I can just remove it. But it's for a maintainer of this existing code to decide, not for a contributor who didn't change the code in question.

Reply:
```suggestion
			if source.AttributeAvatar != "" {
```